### PR TITLE
Add buttons to persist/unpersist a database and a model for PoC

### PR DESCRIPTION
**Pull request by ranquild**
Relies on:
- `card.persisted`
- `database.features["persist-models"]`
- `database.details["persist-models"]`
-  `POST("/api/card/:cardId/persist")`
- `POST("/api/card/:cardId/unpersist")`
- `POST("/api/database/:dbId/persist")`
- `POST("/api/database/:dbId/unpersist")`

Known issues:
- After persiting/unpersisting a model, a page refresh is required for the UI to reflect the `persisted` status

Screenshot 2022-03-30 at 16 08 26

Screenshot 2022-03-30 at 14 55 29

### CODE REVIEW

File: frontend/src/metabase/admin/databases/database.js [hunk 1]
```diff
@@ -283,6 +286,20 @@ export const discardSavedFieldValues = createThunkAction(
...
+  PERSIST_DATABASE,
+  databaseId => async () => {
+    await MetabaseApi.db_persist({ dbId: databaseId });
+  },
+);
+
+export const unpersistDatabase = createThunkAction(
+  UNPERSIST_DATABASE,
+  databaseId => async () => {
+    await MetabaseApi.db_persist({ dbId: databaseId });
```
Comment (dpsutton) [inline frontend/src/metabase/admin/databases/database.js:299]: I'm surprised to see the persist and unpersist both call `MetabaseApi.db_persist({ dbId: databaseId })` i would have expected them to be a little different. Am i missing something? I see the first arg to create ThunkAction is a little different.

Comment (ranquild) [reply frontend/src/metabase/admin/databases/database.js]: Thanks, fixed!

File: frontend/src/metabase/query_builder/components/QuestionActionButtons.jsx [hunk 2]
```diff
@@ -62,6 +71,12 @@ function QuestionActionButtons({
     areNestedQueriesEnabled &&
     checkDatabaseSupportsModels(question.query().database());
 
+  const canPersistDataset =
+    canWrite &&
+    isSaved &&
+    isDataset &&
+    checkDatabaseSupportsPersistence(question.query().database());
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/query_builder/components/QuestionActionButtons.jsx:78]: `checkDatabaseSupportsPersistence` checks that `persist-models` feature is enabled. Or are you talking about a different check?

[the remark this answers is not in the document]
Comment (dpsutton) [reply frontend/src/metabase/query_builder/components/QuestionActionButtons.jsx:78]:
```js
export function checkDatabaseSupportsPersistence(database?: Database | null) {
   return database && database.supportsPersistence();
 }
```

```js
  isPersisted() {
     return this.details["persist-models"];
   }

   supportsPersistence() {
     return this.hasFeature("persist-models");
   }
```

`supportsPersistence` would return true for all postgres and redshift databases, not necessarily ones that had persistence enabled unless i'm misreading this.

**Review by dpsutton — APPROVED** [review]

**Pull request merged**